roles/testnode/files/libexec/smart.sh

```diff
@@ -0,0 +1,285 @@
+#!/bin/bash
+# Description:	Bash script to check drive health using pending, uncorrectable,
+# and reallocated sector count
+#
+# Nagios return codes: 0 = OK; 1 = WARNING; 2 = CRITICAL; 3 = UNKNOWN
+# SMART Attribute Codes:
+#   5 = Reallocated
+#   187 = Reported Uncorrect
+#   197 = Pending
+#   198 = Uncorrectable Sector Count
+#
+# TO-DO: Add support for dynamic SMART attribute lookup.  For example,
+#        187 is reported for Seagate HDD and all SSDs but not Hitachi HDDs.
+#
+# See https://en.wikipedia.org/wiki/S.M.A.R.T.#ATA_S.M.A.R.T._attributes
+
+### Pre-flight checks ###
+# Set raid var then check for cli64 command and bail if missing
+if lspci | grep -qi areca
+then
+  raid=true
+else
+  raid=false
+fi
+
+if [ "$raid" = true ] && ! [ -x "$(command -v cli64)" ]
+then
+  echo "ERROR - cli64 command not found or is not executable"
+  exit 3
+fi
+
+# Check for smartmontools and bail if missing
+if ! [ -x "$(command -v smartctl)" ]
+then
+  echo "ERROR - smartctl is not installed or is not executable"
+  echo "yum/apt-get install smartmontools"
+  exit 3
+fi
+
+### Define global variables ###
+# total number of drives (or RAID slots) discovered
+numdrives=0
+# Number of failed, failing, and/or missing drives
+failingdrives=0
+# Fallback message for UNKNOWN return code output
+unknownmsg="Unknown error"
+# Return code for nagios (Default to UNKNOWN)
+rc=3
+# Array of messages indicating drive health.  Output after nagios status.
+declare -a messages
```

Comment:
that should only be the declare, and you can make it global with -g I think.  experimenting